### PR TITLE
New post: Personal Website featured-project

### PR DESCRIPTION
**Pull request by andrew-chang-dewitt**

### CODE REVIEW

File: src/blog/posts/personal-website.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+In addition to the easy issues, there was one issue involving URL query parameters.
+While I've [written about it previously](/blog/posts/gatsby-query-parameters), the short version is that I wanted an easy way of handling query params (something similar to React's `useState()`) & there was no good solution in Gatsby's router of choice: Reach Router.
+I ended up writing a [custom hook](https://github.com/andrew-chang-dewitt/andrew-chang-dewitt-website/blob/984177220baf696d31fc71fa8ec172475adb2c7e/src/utils/queryHooks.ts) that turned out to be surprisingly useful abstraction making thinking about URL query parameters much simpler.
+In fact, I found a similar need later then when working with React Router, so I've even [modified it](https://github.com/andrew-chang-dewitt/we-cook-sometimes/blob/b1b8a5c125f875dd8a7190d527f879cc066cedb1/src/utils/useQueryParam.ts) it to work with the different Router (along with making some improvements to the typing).
+
+Unresolved issues
+---
+
+In what turned out to be the most frustrating (although largely workable) issue, I often felt like  Gatsby itself was doing too much.
```
Comment (gina40735) [inline src/blog/posts/personal-website.md:104]: It's just my personal preference I guess, but again, I still find "frustrating" too emotional. I'd be worried about your ability in regulating emotions when facing challenges if I were an employer.

File: src/blog/posts/personal-website.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+I'd originally chosen Gatsby one main reason: I wanted an SSG to make creating a multi-page website with dynamic pages simple & performant, and at first, Gatsby seemed to be perfect for this.
```
Comment (gina40735) [inline src/blog/posts/personal-website.md:105]: Did you mean "for" one main reason?

File: src/blog/posts/personal-website.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+Yet the more I worked on the project, the more often I felt like Gatsby was getting in my way.
+
+The most common issue was entirely the caused by one feature of Gatsby that's sold pretty strongly in the tutorial and in the API Reference docs (fittingly filed under ['Gatsby Magic'](https://www.gatsbyjs.com/docs/gatsby-magic/#routing)): it's `Link` component.
```
Comment (gina40735) [inline src/blog/posts/personal-website.md:108]: "that" caused by one feature?

File: src/blog/posts/personal-website.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+---
+
+What & why?
+===
+
+About one year ago, I began studying in earnest with the goal of making a career change to software development.
+As a self-taught developer, I knew I would need a means of proving my skills, so I began to maintain a GitHub account & put together a portfolio with the plan of hosting it on a personal website I'd build myself.
+This summer, I spent some time building the website (if you're reading this, you're probably looking at it now).
+
+While I'm not going to go over how I built every part of the site (most of it's just not that interesting), I definitely benefited from this project & want to discuss what I wanted to achieve by building a website as well as review some of the lessons I learned while building it.
```
[the remark this answers is not in the document]
Comment (andrew-chang-dewitt) [reply src/blog/posts/personal-website.md:22]: parenthetical removed in edits

File: src/blog/posts/personal-website.md (hunk 1 above)
Comment (andrew-chang-dewitt) [reply src/blog/posts/personal-website.md:104]: Changed to 'troublesome', it better conveys what my issue was anyways.

File: src/blog/posts/personal-website.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+Tech stack
+---
+
+When I started, it'd been a few years since I'd done any web development (most of it with Angular 1 & some Aurelia), as I'd been mainly working on IoT services, CLI scripts & programs, & other non-web tech all written in Python & Rust at the time.
+I knew I wanted to build this project as a static site (as much as possible at least), so I began looking at static site generators.
+First, I tried out [Zola](https://www.getzola.org/) because it was written in Rust & looked fairly simple.
+While it worked exactly as advertised, it fell to the same issues I'd had with web development in general back when I'd worked with Angular 1: difficulty writing HTML & CSS in the DRY, composable manner that I'd gotten used to in other areas of software programming.
+
+[Fairly quickly](https://github.com/andrew-chang-dewitt/andrew-chang-dewitt-website/tree/36ed4171044b990741b69fb0e3c55700101c40e5), I abandoned Zola after getting too frustrated with the previously-mentioned difficulty of writing modular markup & styles & a quick peek at Jekyll and similar classic Python SSG's indicated I'd run into the same problems.
```
[the remark this answers is not in the document]
Comment (andrew-chang-dewitt) [reply src/blog/posts/personal-website.md]: Reworded in most recent commit.

**Pull request merged**